### ISSUE TITLE
[Feature] Multi-GPU training

### ISSUE THREAD

**Issue by toby-clark4** [genomicsxai/alphagenome_ft#9]
Hi, thanks for releasing this it's been really useful!

I've been using the package for multi-GPU training and was wondering if you'd be interested in this as a PR?

Currently the adapted training script works with single or multi-GPU, but requires batch size to be set to global batch size (and usually drop_last to be set to True). Would it be best to just update train.py's default training function or add a train_multi_gpu function? The main changes are applying `jax.pmap` along the data axis instead of the current `jax.jit` and some helper functions for batch loading. 

Thanks!

**Comment by Al-Murphy** [comment]
Hey Toby! Yeah feel free to submit a PR - I think of the changes are only as you described, updated train.py to work with both.

Cheers!
Alan.

**Comment by toby-clark4** [comment]
Hey Alan! I think it's basically PR-ready, I've got a test script written for it but there is currently no testing on the `train.py` file. Should I include the testing script in the PR or just provide the updated training script?

Cheers,
Toby

**Comment by Al-Murphy** [comment]
I think just the training script for now, thanks!

**Comment by toby-clark4** [comment]
Have submitted the PR, and let me know if there's anything it'd be helpful to change, thanks!
